### PR TITLE
WIP. Adding the Gitea RepoUrlPicker

### PR DESCRIPTION
**Pull request by cmoulliard**
## Hey, I just made a Pull Request!

PR to enhance backstage in order to propose a Gitea: 
- RepoUrlPicker
- Publish action 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by twanbeeren** [comment]
I don't have access to the repository at the moment to help with this, how can I get this access?

**Comment by cmoulliard** [comment]
> I don't have access to the repository at the moment to help with this, how can I get this access?

I give you access to my forked project @twanbeeren

**Comment by cmoulliard** [comment]
Why did you closed it as no core backstage engineer has been able to look to the PR and also help us on this ? @vinzscam @Rugvip

**Comment by benjdlambert** [comment]
@cmoulliard is this ready for review? :pray:

**Comment by cmoulliard** [comment]
> is this ready for review?

More or less. I have a problem as I cannot really test (as popup list is empty even if locally I configure gitea - why ??) or debug the code change part of this PR. If you can guide me to try to figure out what we should do, that should help us a lot @benjdlambert

**Comment by benjdlambert** [comment]
Hmm @cmoulliard looks like this check is returning `null`. Do you have the required integrations config set?

https://github.com/backstage/backstage/pull/20899/files#diff-9493640b294324e5d442dab8805fcd178cf90dd8dfa070bcbceb474025052cb4R159-R160

**Comment by cmoulliard** [comment]
> Do you have the required integrations config set?

What is a configSet ? What do you expect to have as configSet ? @benjdlambert

**Comment by benjdlambert** [comment]
Sorry I mean `integrations` config like this: https://backstage.io/docs/integrations/gitea/locations/#configuration and is this all set correctly?

**Comment by cmoulliard** [comment]
Here is what I did to test locally: https://gist.github.com/cmoulliard/6e8aa6838e449db42e97ad1f484912f8

**Comment by cmoulliard** [comment]
That works now with my last as I can see the gitea instance running as docker container
Screenshot 2023-12-13 at 12 39 19

End of the scaffolding, I got such an error:
Screenshot 2023-12-13 at 15 13 45

**Comment by benjdlambert** [comment]
@cmoulliard are you running node 20? I think that you will need to pass `NODE_OPTIONS=--no-node-snapshot` [as per the docs](https://backstage.io/docs/getting-started/configuration/#create-a-new-component-using-a-software-template) or try with node 18 instead to test your changes

**Comment by cmoulliard** [comment]
> are you running node 20?

No. I'm using this version` v19.9.0`

**Comment by cmoulliard** [comment]
Still one error to fix after setting the env var `NODE_OPTIONS=--no-node-snapshot`
```
[51](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-51)2023-12-13T16:13:02.000Z Finished step Generating component[52](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-52)2023-12-13T16:13:02.000Z Beginning step Publishing to Source Code Repository[53](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-53)2023-12-13T16:13:02.000Z NotFoundError: Template action with ID 'publish:gitea' is not registered.[54](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-54)    at TemplateActionRegistry.get (/Users/cmoullia/code/janus-idp/fork-backstage/plugins/scaffolder-backend/src/scaffolder/actions/TemplateActionRegistry.ts:39:13)[55](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-55)    at NunjucksWorkflowRunner.executeStep (/Users/cmoullia/code/janus-idp/fork-backstage/plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:234:37)[56](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-56)    at NunjucksWorkflowRunner.execute (/Users/cmoullia/code/janus-idp/fork-backstage/plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:433:9)[57](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-57)    at TaskWorker.runOneTask (/Users/cmoullia/code/janus-idp/fork-backstage/plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts:145:26)[58](http://localhost:3000/create/tasks/f3b616c4-9744-4e68-bb1f-966a6a5f789c#line-58)    at run (/Users/cmoullia/code/janus-idp/fork-backstage/node_modules/p-queue/dist/index.js:163:29)
```

**Comment by benjdlambert** [comment]
Hmm but I think that this is expected though, as there is no `publish:gitea` action I believe. I think that there's still some more work to do to enable gitea support for the scaffolder end to end, but this is a good step in the right direction 👍 

We need a `publish:gitea` action that mirrors the `publish:github` and `publish:gitlab` action, that can create the repository in Gitea and then push the current `workdir`.

**Comment by cmoulliard** [comment]
> We need a `publish:gitea` action

Ok. I know what to do now as additional step ;-)

**Comment by benjdlambert** [comment]
@cmoulliard no I think that this is fine to ship separately. You will need changesets though.

You can run `yarn changeset` and select the affected packages.

File: .changeset/giant-pets-cover.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': major
```
Comment (benjdlambert) [inline .changeset/giant-pets-cover.md:2]:
```suggestion
'@backstage/plugin-scaffolder': minor
```

This should be a `minor` release as there's no breaking changes here, just a feature addition :)

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just a small thing around the changeset

File: .changeset/giant-pets-cover.md (hunk 1 above)
Comment (cmoulliard) [reply .changeset/giant-pets-cover.md]: Fixed with bae1e40

**Comment by benjdlambert** [comment]
Ah one last thing - the build is failing. You might have to run `yarn build:api-reports` and commit all api-report.md changes.

**Review by benjdlambert — APPROVED** [review]
Nice thanks :tada:

**Pull request merged**